lib/KCard/index.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+
+  <BaseCard
+    :title="title"
+    :headingLevel="headingLevel"
+    :titleLines="titleLines"
+    :to="to"
+  >
+    <template #default>
+      <aside
+        v-if="layout === 'vertical' && thumbnailDisplay !== 'none'"
+      >
+        <KImg
+          :src="thumbnailSrc"
+          :isDecorative="true"
+          :style="KImgVerticalSmallStyle"
+        />
+        <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+      </aside>
+
+      <div class="horizontal-layout-style">
+        <aside
+          v-if="layout === 'horizontal' && thumbnailDisplay === 'large'"
+        >
+          <KImg
+            :src="thumbnailSrc"
+            :isDecorative="true"
+          />
+          <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+        </aside>
+
+        <div>
+          <header
+            v-if="thumbnailDisplay === 'none'"
+            class="spacing"
+          >
+            <!-- @slot Places content to be placed above the title.. -->
+            <slot name="aboveTitle"></slot>
+            <!-- @slot for the title content -->
+            <slot name="title"></slot>
```

Comment:
I think this title slot is an alternative to the title prop. So I think we will probably want to make this part of the BaseCard component perhaps?

Reply:
That would not be a good idea. I think `BasCard` is to enforce consistent structure and accessibility for cards across  our product and Isolates basic accessibility requirements from potential changes in the `KCard` component,

Reply:
`BaseCard` is meant to have the validation _"Contains validation to check that title prop or slot has been used, throwing an error if neither is provided"_ (see acceptance criteria here https://github.com/learningequality/kolibri-design-system/issues/529) so I think the slot will need to be present in the `BaseCard`. It will add some clarity as well. This is part of the enforcement that `BaseCard` does - it needs to ensure that title is provided, and that can happen either via the prop or the slot.

@AllanOXDi you too have made a good point and regarding that, we need to ensure that `BaseCard` shouldn't have any styles or layout logic related to how the title appears.